### DISCUSSION TITLE
Differentiate between self hosted/main & Sentry

### DISCUSSION THREAD

**Discussion by tomlawesome** [lukevella/rallly#137] [General]
Hello,

Love Rallly, it's great functionally. Thanks for your time and effort, will hopefully help with the days long WhatsApp chats with mates to sort out getting together ona day we're all free!

It would be great though if there was perhaps a separate repo for self-hosting that had an alternative landing page rather than essentially re-hosting rallly.co. I don't mean like a white label, just I think most people who self-host would only want to go direct to a 'new poll' page every time and not have a full website around it.

I do of course appreciate this can absolutely be done if you understand Tailwind CSS etc, but I'm a web-dev noob. 

I also noticed that the container includes sentry.io. Is this sending telemetry back to your Sentry account about the app and its usage etc? If so, it might be worth making it clear what data you can/can't see.

**Comment by lukevella** [comment]
Hi @tomlawesome. I completely agree, a separate project for self-hosting makes sense. Same with having the poll creation page be the default page. Totally makes sense to me. The only reason it's not this way is because it was quicker/easier to just keep everything in one project to start but you can expect to see this change.

> I do of course appreciate this can absolutely be done if you understand Tailwind CSS etc, but I'm a web-dev noob.

To be honest all you need to do is modify [this line in next.config.js](https://github.com/lukevella/rallly/blob/main/next.config.js#L33)

```diff
- destination: "home"
+ destination: "new"
```

And the root page will be the poll creation page. Not recommending you do this as it makes it harder to pull updates from the repo but just saying.

With regards to sentry, nothing is sent back unless a [`SENTRY_AUTH_TOKEN` is defined](https://github.com/lukevella/rallly/blob/main/next.config.js#L61). It's not intended to be used for users who are self-hosting since it would require being able to modify the `sentry.properties` file in the project.

**Comment by tomlawesome** [comment]
Awesome, thanks will give that a go as it’s a one liner! I have in the past span off my own private version and found it’s more hassle than it’s worth to later try and incorporate updates. Would much rather leave vanilla and pull repo/rebuild container. 😅

Good news on the sentry stuff, totally makes sense.

**Comment by SF2311** [comment]
Hello,
thank you for the great Software.

I would like to host a rallly instance for an organization and we would ideally like the following features:
* Customizable logo and link to homepage
* Customizable Footer Menu (displayed links and their target)
* Customizable Social Media links (which ones should be displayed and their target)

Is this something you would consider adding to the project, or accepting if we were to develop it?

**Reply by lukevella** [reply]
Hi @SF2311, I do plan on allowing for more customization for self-hosted instances but it's not what I'm actively working on at the moment so I'm not able to provide an estimate on when these will be available. I am working on a major update to the product so it will be unlikely that any third-party work done here will be mergeable since it would be too time-consuming to resolve the inevitable conflicts that would occur from working on two very different versions of the codebase.

**Reply by hollosch** [reply]
@lukevella 
Thanks for the information. We would also like to personalize the start page for our club and are patiently looking forward to it!

**Comment by AnthonyFR-ACE** [comment]
Hello, I'm re-launching the subject for my association, thank you for your work.
